core/src/main/java/com/graphhopper/coll/GHSortedCollection.java

```diff
@@ -105,14 +105,21 @@ public int pollKey() {
         if (size < 0) {
             throw new IllegalStateException("collection is already empty!?");
         }
-        Entry<Integer, TIntHashSet> e = map.firstEntry();
-        TIntHashSet set = e.getValue();
-        TIntIterator iter = set.iterator();
+
+        Entry<Integer, GHIntHashSet> e = map.firstEntry();
+        GHIntHashSet set = e.getValue();
         if (set.isEmpty()) {
             throw new IllegalStateException("internal set is already empty!?");
         }
-        int val = iter.next();
-        iter.remove();
+
+        Iterator<IntCursor> iter = set.iterator();
+        final int val = iter.next().value;
+        set.removeAll(new IntPredicate() {
```

Comment:
Why not just use http://carrotsearch.github.io/hppc/releases/0.7.0/api/com/carrotsearch/hppc/IntHashSet.html#removeAll-int- ?